rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp

```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
             replicate_last_batch_to_pad_partial_shard();
         }
     }
-    //shuffle dataset if set
-    if (ret == Reader::Status::OK && _shuffle)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+
+    if (_meta_data_reader && _meta_data_reader->aspect_ratio_grouping())
+    {
+        for (const auto &filename : _file_names)
+        {
+            std::string base_filename = filename.substr(filename.find_last_of("/\\") + 1);
+            auto img_size = _meta_data_reader->lookup_image_size(base_filename);
+            auto aspect_ratio = static_cast<float>(img_size.h) / img_size.w;
+            _aspect_ratios.push_back(aspect_ratio);
+        };
+
+        // zip the two vectors
+        std::vector<std::pair<std::string, float>> zipped(_file_names.size());
+        for (size_t i = 0; i < _file_names.size(); i++)
+            zipped[i] = std::make_pair(_file_names[i], _aspect_ratios[i]);
+
+        // sort according to aspect ratios
+        std::sort(zipped.begin(), zipped.end(), [](auto &lop, auto &rop)
+                  { return lop.second < rop.second; });
+
+        // extract sorted file_names
+        std::vector<std::string> sorted;
```

Comment:
Rename this to something more readable like sorted_file_names instead of just sorted